src/v/syschecks/syschecks.cc

```diff
@@ -48,24 +50,42 @@ void memory(bool ignore) {
     }
 }
 
-void systemd_notify_ready() {
+ss::future<> systemd_notify_ready() {
     ss::sstring msg = fmt::format(
       "READY=1\nSTATUS=redpanda is ready! - {}", redpanda_version());
-    systemd_raw_message(msg);
+    return systemd_raw_message(std::move(msg));
 }
 
-// TODO:agallego - support non-systemd installations
-// by adding a macro around sd_notify to disable at compile time
-void systemd_raw_message(const ss::sstring& out) {
-    auto r = sd_notify(0, out.c_str());
+ss::future<> systemd_raw_message(ss::sstring out) {
+    const char* systemd_socket_path = std::getenv("NOTIFY_SOCKET");
+    std::string_view log_msg = out;
     if (out.back() == '\n') {
-        checklog.debug("{}", std::string_view(out.c_str(), out.size() - 1));
-    } else {
-        checklog.debug("{}", out);
+        // emit tight logs without extra new lines
+        log_msg = std::string_view(out.c_str(), out.size() - 1);
+    }
+    if (!systemd_socket_path) {
+        checklog.trace("NOTIFY_SOCKET unset. ignoring {}", log_msg);
+        co_return;
+    }
+    checklog.debug("{}", log_msg);
+    ss::sstring systemd_socket = systemd_socket_path;
+    if (systemd_socket[0] == '@') {
+        // detected abstract socket; replace @ with 0
+        systemd_socket[0] = '0';
     }
-    if (unlikely(r < 0)) {
-        checklog.trace("Could not notify systemd sd_notify ready, error:{}", r);
+    auto sock = co_await ss::engine().net().connect(
```

Comment:
Does it make sense to cache this connection?

Reply:
i thought about it, but it's used only on the first couple of seconds, seems OK to me. but also... just found out 

```
Error sending systemd notification: sendto: Address family not supported by protocol```


We need to add support for unix-sockets + UDP in seastar. it only supports tcp + unix.

Reply:
I'm sure it's quick, and it isn't used much or on the critical path, 'twas more of an observation.

Reply:
@senior7515 so we gonna use normal blocking sockets here?

Reply:
@dotnwat yup. is just one data gram. i just updated it.